js/copy-service/Parser/Parser.ts

```diff
@@ -523,14 +509,14 @@ class Parser {
       }
 
       const parsed = isRestricted ?
-        this._parseTokens(parsedOptionalArgumentsTokens, key, true, expectedEndingToken) :
-        this._parseTokens(parsedOptionalArgumentsTokens, key);
+        this._parseTokens(parsedOptionalArgumentsTokens!, key, true, expectedEndingToken) :
+        this._parseTokens(parsedOptionalArgumentsTokens!, key);
 
       return {
         ast: new Functional({
           copy: firstParsed.ast,
           key: textParsed.text,
-          args: _.get(argumentsParsed, 'args'),
+          args: _.get(argumentsParsed, 'args')!,
```

Comment:
This could be `undefined` if the branch at line 504 is not taken, correct? I think `args` should be optional on Functional, especially given we default `args` to `[]` in the constructor. https://github.com/NextCapital/copy-service/blob/main/js/copy-service/Functional/Functional.ts#L39